gnd/src/main/java/com/google/android/gnd/persistence/local/room/entity/FeatureEntity.java

```diff
@@ -160,14 +164,28 @@ public static Feature toFeature(FeatureEntity featureEntity, Project project) {
   }
 
   public static String listToString(ImmutableList<Point> vertices) {
+    ArrayList<ArrayList<Double>> polygonVertices = new ArrayList<>();
+    for (Point pt : vertices) {
+      ArrayList<Double> innerValues = new ArrayList<>();
```

Comment:
General guidance:
* Please give variables more meaningful names, i.e., `coordinate`.
* Declare variables use most generic interface needed, which in this case would be `List`.

That said, this code and the code below is hard to follow and can be greatly simplified. For example, this entire method can be replace with:

```java
public static String listToString(List<Point> vertices) {
  List<List<Double>> verticesArray = stream(vertices).map(point => ImmutableList.of(point.getLatitude(), point.getLongitude)).collect(ImmutableList.toImmutableList());
  return gson.toJson(verticesArray);
}
```

Reply:
method updated with  
````
     List<List<Double>> verticesArray = stream(vertices)
        .map(point -> ImmutableList.of(point.getLatitude(),
            point.getLongitude())).collect(
          Collectors.toList());
    return gson.toJson(verticesArray);
````